README.md

```diff
@@ -28,16 +28,22 @@ go get github.com/casbin/casbin-dashboard
 
 Casbin-dashboard will store its metadata in a MySQL database named: `casbin_metadata`, will create it if not existed. The DB connection string can be specified at: https://github.com/casbin/casbin-dashboard/blob/master/conf/app.conf
 
+- Setup casbin-dashboard to enable some third-party login platform
+
+Casbin-forum provide a way to sign up using Github account, so you may have to get your own `GithubAuthClientID`, `GithubAuthClientSecret` first.
+
+You could get them by clicking on this url: https://github.com/settings/developers , You should set `Homepage URL` to fit your own domain address, for local testing, set`http://localhost:3000`. And set the Authorization callback URL, the same domain address as `Homepage URL`, for local testing, set`http://localhost:3000`.

```

Comment:
the callback URL should be `http://localhost:3000/callback/signup`

Reply:
The reason why I set `http://localhost:3000` to be the callback URL for locally test is that I have made the callback Url in
[setting.js](https://github.com/casbin/casbin-dashboard/blob/235323eefd785cd85d8422567e45abd3bd1f5703/web/src/Setting.js#L41).

Reply:
In the Setting.js file, the `redirect_uri`  in query parameter is  format as`${ClientUrl}/callback/${method}` . In that case, the request call for GitHub Auth Api will generate a query parameter like is `localhost:3000/callback/signup` if the domain is set as `localhost`

 But according to the GitHub doc , [reference](https://docs.github.com/en/free-pro-team@latest/developers/apps/troubleshooting-authorization-request-errors#redirect-uri-mismatch)

> If you provide a `redirect_uri` that doesn't match what you've registered with your application, GitHub will redirect to the registered callback URL with the following parameters summarizing the error
>
> ```
> http://your-application.com/callback?error=access_denied
>   &error_description=The+user+has+denied+your+application+access.
>   &error_uri=/apps/building-integrations/setting-up-and-registering-oauth-apps/troubleshooting-authorization-request-errors/%23access-denied
>   &state=xyz
> ```

Reply:
Using `http://localhost:3000` to be the callback URL will not lead to mismatch problem. The redirect URL's host and port must exactly match the callback URL and must reference a `subdirectory` of the callback URL.
[reference](https://docs.github.com/en/free-pro-team@latest/developers/apps/authorizing-oauth-apps#redirect-urls)

So both `http://localhost:3000` and `http://localhost:3000/callback/signup` can be used here, but I mentioned if the `method` changed, using `http://localhost:3000/callback/signup` as callback URL may lead to mismatch problem. [Setting.js](https://github.com/casbin/casbin-dashboard/blob/235323eefd785cd85d8422567e45abd3bd1f5703/web/src/Setting.js#L41).

Reply:
![Screenshot from 2021-01-05 00-09-10](https://user-images.githubusercontent.com/34754265/103568057-e4a0f680-4eea-11eb-9e99-cc1297feb185.png)
even after setting the configurations for the **Homepage URL** and **Callback URL** to `http://localhost:3000`, I am getting the above error 
@Xiaoyang-Gu can you help?

Reply:
@Xiaoyang-Gu

Reply:
@pranav1698 I recommend you to fix the authentication header. [link](https://github.com/casbin/casbin-dashboard/pull/136)

Reply:
@Xiaoyang-Gu can you please tell me where and what changes needs to be done in the master branch?
Still, can't figure out what is the reason for this error?

Reply:
@pranav1698 Can you show me your console log or other error message? I cannot get what error you meet.

Reply:
@Xiaoyang-Gu changed the port to 3030 and it worked
![Screenshot from 2021-01-06 23-03-16](https://user-images.githubusercontent.com/34754265/103801218-6de33500-5073-11eb-9744-4e9fbc52314f.png)

Reply:
OK